frontend/lib/src/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -263,38 +268,35 @@ export const NumberInput: React.FC<Props> = ({
     ]
   )
 
-  const onBlur = (): void => {
+  const onBlur = useCallback((): void => {
     if (dirty) {
       commitValue({ value, source: { fromUi: true } })
     }
     setIsFocused(false)
-  }
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [dirty])
```

Comment:
if we don't pass `value` as a dependency, it might be possible that we commit an old value. Does anything speak against including this and `commitValue` (which is a stable reference because of `useCallback`)?